Core/base elements content only (#1299)
massgov/mayflower: packages/core/stories/elements/Button/Button.stories.mdx

/>

# Buttons

> Buttons are used in forms and interactive applications, but also as prominent call-to-action links on website pages.

Reviewer: So here I'm thinking whether we should make the distinction between buttons and links. In the Link doc, there's a variant called "Link styled as a button", meaning button "styles" are allowed to be used as links (whether or not it's a good practice it's up for debate).  Technically, a button should only be used to trigger actions vs links are used for navigation. 
```suggestion
Buttons are used in forms and interactive applications to trigger actions.

Reviewer: What I'm trying to say here is whether this doc is about the element button or the styles of buttons — if we are keeping "button" just as the element itself, then I think to distinguish it from links will make more sense.

<Canvas withSource="none">
  <Story
		name="button-primary"
    parameters={{ docs: { transformSource: () => renderReactMarkup(<Button usage="" text="Button"/>) } }}>
      <Button usage="" text="Button" />
  </Story>
</Canvas>

Following web best practices buttons have a target area of 45 pixels. To provide more visual affordance that they are clickable—not just text in a box—they have a shadow to appear slightly raised. Small buttons can be used for secondary tasks on forms—for example image upload—and filters on listing pages. Large buttons should be used for form submits or primary calls to action. There should usually be only one primary button in any UI. Font sizes reduce slightly on mobile.

---

### Primary Buttons

- **Usage:** For the main action in a form, or for a call-to-action link on a page.

> - **Style:** Default buttons are solid color with contrasting uppercase text. For actions with text longer than two words use the `title-case` class. On hover the Background color becomes lighter.

Reviewer: so the "title-case" option is not currently available to Button, it is though in ButtonWithIcon in react (but not in patternlab), which we should make available to all buttons. These are the things I was talking about that we need to follow up on refactoring and consolidating in code. 
```suggestion
- **Style:** Default buttons are solid color with contrasting uppercase text. For actions with text longer than two words use the title case instead of all caps. On hover, the Background color becomes lighter.
```

Reviewer: for now, a class is not available, so let's just describe the rule (and add that as a to-do in code)

<Canvas>
  <Story
    name="button-secondary-c-primary"
    parameters={{ docs: { transformSource: () => renderReactMarkup(<Button usage="secondary" text="Secondary"/>) } }}>
      <Button usage="secondary" text="Secondary" />
  </Story>
  <Story
    name="button-secondary-c-primary-alt"
    parameters={{ docs: { transformSource: () => renderReactMarkup(<Button usage="secondary" theme="c-primary-alt" text="Secondary"/>) } }}>
      <Button usage="secondary" theme="c-primary-alt" text="Secondary"/>
  </Story>
  <Story
    name="button-secondary-c-highlight"
    parameters={{ docs: { transformSource: () => renderReactMarkup(<Button usage="secondary" theme="c-highlight" text="Secondary"/>) } }}>
      <Button usage="secondary" theme="c-highlight" text="Secondary"/>
  </Story>
  <Story
    name="button-secondary-c-gray"
    parameters={{ docs: { transformSource: () => renderReactMarkup(<Button usage="secondary" theme="c-gray-dark" text="Secondary"/>) } }}>
      <Button usage="secondary" theme="c-gray-dark" text="Secondary"/>
  </Story>
  <Story story={stories.buttonCSS2} />
</Canvas>

---

### Tertiary Buttons

> - **Usage:** For third-level actions on a page.

Reviewer: ```suggestion
- **Usage:** For third-level actions.
```

> - **Style:** The gray color is even less prominent than outline button to de-emphasize third-level tasks. The hover state inverts the background and text colors.

Reviewer: ```suggestion
- **Style:** The gray color is even less prominent than outline button to de-emphasize third-level tasks. Background color of the primary button lightens when being hovered over.
```

Sorry, these are the things I missed in the first draft.

Author: Ah, right. I would maybe word it a bit differently:
"The hover state lightens the primary button background color."

Reviewer: oh sorry, this is not the primary button, but the hover animation is similar to the primary button, so I just copied from the primary. This is actually the tertiary button

### Primary Buttons

- **Usage:** For the main action in a form, or for a call-to-action link on a page.

> - **Style:** Default buttons are solid color with contrasting uppercase text. THe hover state lightens the primary button background color.

Author: For consistency with the wording below.

### Tertiary Buttons

- **Usage:** For third-level actions.

> - **Style:** The gray color is even less prominent than outline button to de-emphasize third-level tasks. The hover state lightens the primary button background color.

Reviewer: I guess should be tertiary instead

### Quaternary Buttons

- **Usage:** For fourth-level actions or actions.

> - **Style:** Appears as a link, with a small amount of letterspacing and underline (Actually a `border-bottom` not a text underline). The hover state darkens the underline and increases it's weight.

Author: We can't say this because they would need to create an override to the button class which has text-transform: uppercase to even do it, plus we'd also want to remove the letterspacing. Can we just add a global helper class to Mayflower scss like:
```
.ma__case-title {
  text-transform: capitalize;
  letter-spacing: 0;
}
```

Author: If we do that ^^ it would probably also make sense to do: 
```
.ma__button--quaternary {
-  text-transform: none;
+  text-transform: capitalize;
}

Reviewer: lets do this separately, as this will impact other implementations and create visual regressions for react. we should add a way to config casing consistently across all buttons, currently it's not consistent (This is currently in the to-do list --> https://docs.google.com/document/d/1HEep0C1ySxiNKM3r5oIMHjaAqJN4KXnPlg2NVImOw4g/edit#heading=h.5jchh3ntd4nt)
